plugins/include/messages.inc

```diff
@@ -18,88 +18,587 @@
+ *       it outside of these functions will crash the server immediately.
+ *
+ * @param x           Coordinate to write
+ *
+ * @noreturn
+ */
 native ewrite_coord(x);
+
+/**
+ * Writes a coordinate entry to a message using a float value.
+ *
+ * @note This function should only be used in between a emessage_begin()
+ *       or emessage_begin_f() and a emessage_end() function. Trying to use
+ *       it outside of these functions will crash the server immediately.
+ *
+ * @param x           Coordinate to write
+ *
+ * @noreturn
+ */
 native ewrite_coord_f(Float:x);
+
+/**
+ * Writes a string to a message.
+ *
+ * @note This function should only be used in between a emessage_begin()
+ *       or emessage_begin_f() and a emessage_end() function. Trying to use
+ *       it outside of these functions will crash the server immediately.
+ *
+ * @param x           String to write
+ *
+ * @noreturn
+ */
 native ewrite_string(const x[]);
 
-/* Sets/Gets what engine messages are blocked. */
+/**
+ * Sets whether or not an engine message will be blocked.
+ *
+ * @note For a list of message flags, have a look at the BLOCK_* constants
+ *       in message_const.inc.
+ *
+ * @param iMessage       Message id
+ * @param iMessageFlags  BLOCK_* constant
+ *
+ * @noreturn
+ * @error                If an invalid message id is specified, an error
+ *                       will be thrown.
+ */
 native set_msg_block(iMessage, iMessageFlags);
+
+/**
+ * Gets whether or not an engine message is blocked.
+ *
+ * @param iMessage       Message id
+ *
+ * @return               BLOCK_* constant
+ * @error                If an invalid message id is specified, an error
+ *                       will be thrown.
+ */
 native get_msg_block(iMessage);
 
-/* Lets you directly hook a message in the engine!
- * You can overwrite the message before anything happens and either let the message continue
- * or fully block it. Here is how it works:
- * If you hook a message, the message is stored but not sent. You have the opportunity to
- * not only execute code, but to get/set the contents of the message, before you choose to 
- * either block it or let it go on its way. The hooked function will be passed a msg_id, msg_dest, and entity index. 
- * The return value can be passed to unregister_message() to stop the message from being hooked */
+/**
+ * Lets you directly hook a message in the engine.
+ *
+ * @note The function is called in the following manner:
+ *   msg_id          - Message id
+ *   msg_dest        - Destination type (see MSG_* constants in messages_const.inc)
+ *   msg_entity      - Entity receiving the message
+ *
+ * @note You can overwrite the message before anything happens by using the
+ *       set_msg_arg_* functions and either let the message continue by
+ *       returning PLUGIN_CONTINUE or fully block it with PLUGIN_HANDLED.
+ * @note If you hook a message, the message is stored but not sent. You have
```

Comment:
Maybe we should add a reference about `register_event` here? Something like: "It is the opposite to register_event. register_event executes your code after the message is sent."